src/org/labkey/targetedms/passport/PassportController.java

```
@@ -0,0 +1,395 @@
+                "r.dataid, r.filename, r.created, r.modified, r.formatversion " +
+                "FROM targetedms.peptidegroup pg, targetedms.runs r, prot.sequences ps " +
+                "WHERE r.id = pg.runid AND r.container = ? AND ps.seqid = pg.sequenceid " +
+                "AND pg.accession = ? LIMIT 1;");
+        targetedMSProteinQuery.add(getContainer().getId());
+        targetedMSProteinQuery.add(accession);
+        UserSchema schema = QueryService.get().getUserSchema(getUser(), getContainer(), "targetedms");
+        try {
+            Map<String, Object> map = new SqlSelector(schema.getDbSchema(), targetedMSProteinQuery).getMap();
+
+            IProtein p = new IProtein();
+            if(map == null)
+                return null;
+            p.setLabel((String) map.get("label"));
+            p.setGene((String) map.get("bestgenename"));
+            p.setSpecies((String) map.get("species"));
+            p.setPreferredname((String) map.get("preferredname"));
+            p.setPepGroupId((Integer) map.get("pgid"));
+            p.setSequenceId((Integer) map.get("seqid"));
+            p.setDescription((String) map.get("description"));
+            p.setSequence((String) map.get("protsequence"));
+            p.setLength((Integer) map.get("length"));
+
+            IFile f = new IFile();
+            f.setFileName((String) map.get("filename"));
+            f.setSoftwareVersion((String) map.get("softwareversion"));
+            f.setCreatedDate((Date) map.get("created"));
+            f.setModifiedDate((Date) map.get("modified"));
+            f.setRunId((Integer) map.get("runid"));
+
+            p.setFile(f);
+            p.setAccession(accession);
+            populateProteinKeywords(p);
+            populateUniprotData(p);
+            populatePeptides(p);
+//            populateProjects(p); TODO: project stuff needs more work - will happen next update
+            return p;
+        }
+        catch(Exception e) {
+            return null;
```

Don't catch and swallow exception